source/boundary_heat_flux/function.cc

```diff
@@ -0,0 +1,178 @@
+/*
+  Copyright (C) 2011 - 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/boundary_heat_flux/function.h>
+#include <aspect/global.h>
+#include <deal.II/base/signaling_nan.h>
+
+namespace aspect
+{
+  namespace BoundaryHeatFlux
+  {
+    template <int dim>
+    std::vector<Tensor<1,dim> >
+    Function<dim>::
+    heat_flux (const types::boundary_id /*boundary_indicator*/,
+               const MaterialModel::MaterialModelInputs<dim> &material_model_inputs,
+               const MaterialModel::MaterialModelOutputs<dim> &/*material_model_outputs*/,
+               const std::vector<Tensor<1,dim> > &normal_vectors) const
+    {
+      const unsigned int n_evaluation_points = material_model_inputs.position.size();
+      std::vector<Tensor<1,dim> > heat_flux(normal_vectors);
+
+      for (unsigned int i=0; i<n_evaluation_points; ++i)
+        {
+          const Point<dim> position = material_model_inputs.position[i];
+          if (coordinate_system == Utilities::Coordinates::cartesian)
+            {
+              heat_flux[i] *= boundary_heat_flux_function.value(position);
+            }
+          else if (coordinate_system == Utilities::Coordinates::spherical)
+            {
+              const std_cxx11::array<double,dim> spherical_coordinates =
+                aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(position);
+              Point<dim> point;
+
+              for (unsigned int d=0; d<dim; ++d)
+                point[d] = spherical_coordinates[d];
+
+              heat_flux[i] *= boundary_heat_flux_function.value(point);
+            }
+          else if (coordinate_system == Utilities::Coordinates::depth)
+            {
+              const double depth = this->get_geometry_model().depth(position);
+              Point<dim> point;
+              point(0) = depth;
+
+              heat_flux[i] *= boundary_heat_flux_function.value(point);
+            }
+          else
+            {
+              AssertThrow(false, ExcNotImplemented());
+              return std::vector<Tensor<1,dim> > (n_evaluation_points,numbers::signaling_nan<Tensor<1,dim> >());
```

Comment:
is this line needed to compile? I would guess returning heat_flux below is enough.